pkg/controller/operators/catalog/operator.go

```diff
@@ -1196,6 +1257,8 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 
 	logger.Info("syncing")
 
+	o.gcInstallPlans(logger, plan.GetNamespace())
```

Comment:
I think performing this GC on namespace resolve sync would result in less overall churn. Is there a reason to perform this for every InstallPlan event?

Reply:
I reasoned that this hedges against issues with catalogs / the resolver pinning the namespace sync for too long. 

But lots of good ideas here + in other comments, I'll update this PR.